lib/core/request.js

```diff
@@ -329,10 +336,14 @@ function processHeader (request, key, val) {
     throw new NotSupportedError('expect header not supported')
   } else if (tokenRegExp.exec(key) === null) {
     throw new InvalidArgumentError('invalid header key')
-  } else if (headerCharRegex.exec(val) !== null) {
-    throw new InvalidArgumentError(`invalid ${key} header`)
   } else {
-    request.headers += `${key}: ${val}\r\n`
+    if (Array.isArray(val)) {
+      for (let i = 0; i < val.length; i++) {
```

Comment:
Don't we need to check the types of the values in the array?

Reply:
How would you like to think about "types" generally? When we took out the string | number | boolean typecheck above, it opens the door to Symbols, functions, and bigints. 

What may make more sense is to collapse line 282-336 into `processHeaderValue` and have all value type checking there. But then that would bring us back to what types should / should not throw.

Should we take the stance:
* Don't be dumb and we'll only throw an error if the value is `typeof === 'object'`

OR

* We'll throw an error any time value is not a string or string equivalent (string | number | boolean)

?

Reply:
Just check for object like already do.